fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerMutualTest.java

```diff
@@ -976,4 +978,196 @@ void testMultiTargetMultiType() {
         });
         validateIndexes(indexes);
     }
+
+    @Test
+    void testMutualIndexingBlocker() {
+        // build indexes, pause by indexing stamp blocker, release unblock
+        List<Index> indexes = new ArrayList<>();
+        // Here: Value indexes only
+        indexes.add(new Index("indexA", field("num_value_2"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        indexes.add(new Index("indexB", field("num_value_3_indexed"), IndexTypes.VALUE));
+        indexes.add(new Index("indexC", field("num_value_unique"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS));
+        // Here: Add a non-value index
+        indexes.add(new Index("indexD", new GroupingKeyExpression(EmptyKeyExpression.EMPTY, 0), IndexTypes.COUNT));
+
+        openSimpleMetaData();
+        int numRecords = 333;
+        populateData(numRecords);
+
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = allIndexesHook(indexes);
+        openSimpleMetaData(hook);
+        disableAll(indexes);
+
+        // Start indexing and crash, block indexing
+        final List<Tuple> boundariesList = getBoundariesList(numRecords, 4);
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        IntStream.rangeClosed(0, 4).parallel().forEach(id -> {
+            if (id == 3) {
+                FDBRecordStoreTestBase.RecordMetaDataHook localHook = allIndexesHook(indexes);
+                openSimpleMetaData(localHook);
+                try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                        .setDatabase(fdb).setMetaData(metaData).setSubspace(subspace)
+                        .setTargetIndexes(indexes)
+                        .build()) {
+                    Thread.sleep(500); // this sleep is required to make sure that indexing had started and the indexes contain type stamps
+                    indexer.indexingStamp(OnlineIndexer.IndexingStampOperation.BLOCK, null, 0);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            } else {
+                oneThreadIndexingCrashHalfway(indexes, timer, boundariesList, 1);
+            }
+        });
```

Comment:
It seems like we should assert here that none of the indexes are built.

Reply:
✅